contract_api/mpe.py

```diff
@@ -0,0 +1,67 @@
+import web3
+from common.constant import NETWORKS
+from common.utils import Utils
+from contract_api.registry import Registry
+from web3 import Web3
+
+
+class MPE:
+    def __init__(self, net_id, obj_repo):
+        self.repo = obj_repo
+        self.objSrvc = Registry(obj_repo=obj_repo)
+        self.ws_provider = NETWORKS[net_id]['ws_provider']
+        self.obj_util = Utils()
+
+    def get_latest_block_no(self):
+        w3Obj = Web3(web3.providers.WebsocketProvider(self.ws_provider))
+        return w3Obj.eth.blockNumber
+
+    def get_channels_by_user_address(self, user_address, org_id=None, service_id=None):
+        try:
+            print('Inside get_channel_info::user_address',
+                  user_address, '|', org_id, '|', service_id)
+            last_block_no = self.get_latest_block_no()
+            params = [last_block_no]
+            sub_qry = ""
+            if user_address is not None and service_id is not None and org_id is not None:
+                sub_qry = " AND S.org_id = %s AND S.service_id = %s "
+                params.append(org_id)
+                params.append(service_id)
+            params.append(user_address)
+            raw_channel_dta = self.repo.execute(
+                'SELECT  C.*, E.*, IF(C.expiration > %s, "active","inactive") AS status FROM '
```

Comment:
Whats the significance of a channel being active or inactive?

Reply:
We placed an active/inactive status to indicate its expiry status.

Reply:
active or inactive tell us about its expiry status.